lock-impl/src/main/java/com/palantir/lock/impl/LegacyTimelockService.java

```diff
@@ -176,6 +176,9 @@ public long currentTimeMillis() {
         return lockService.currentTimeMillis();
     }
 
+    @Override
+    public void close() {}
```

Comment:
confirming: are we sure we don't want this to close the lock and timestamp services?

Reply:
I suspect at least lock is closed separately, though (1) I would think close is idempotent, and (2) if we're not doing this I think it's worth a comment.

Reply:
none of the lock and timestamp services are closeable, so should be fine to do nothing here.

Reply:
that's not true! legacy timelock service can receive a local lock service - it's hidden behind layers and layers - and it has executors e.g. see `LockServiceImpl`

Reply:
I was trying to say `LockService` interface is not closeable, so `LegacyTimelockService` won't be able to close underlying services - and were not closing them before this pr.
I guess your concern is marking TimelockService as closeable would give the impression that closing it would actually close underlying resources also (which is a pretty valid assumption), and may cause bugs in future, even though it is not changing current behaviour. Is that right? @felixdesouza @jeremyk-91

Reply:
Yes. Part of me feels like if this is a layer on top, then we will never think to close the underlying layers. i.e. I think this change without doing due diligence on the lower layers does more harm than good in the long term.

In my mind, I think the correct approach is what you employed fairly recently: you have a client that's solely responsible for RPC, and it doesn't wrap anything. Then have a corresponding class that may or may not extend the RPC interface (don't care at this point), but it *is* closeable. Otherwise, this will be impossible to track imo.

If you have this model, we don't necessarily have infinite wrappers, we can just have one class that implements the non-rpc "client" and that can customise behaviour however it pleases. We don't have to worry and think: "oh what timestamp service do I actually have here, is the ordering correct?" etc.

As an aside/rambling at this point, the fact we have all of these interfaces are a lie, they're not necessarily interchangeable. I'd rather have something that combined this state and presented it as is, (it can do nesting/wrapping inside the class if it needs to).

Reply:
that makes sense, I guess marking `TimelockService` as closeable is not the correct approach here, given all we wanted was registering close callback of `RemoteTimelockServiceAdapter`. Opened another pr doing that: #3847